genesyscloud/responsemanagement_library/genesyscloud_responsemanagement_library_proxy.go

```diff
@@ -0,0 +1,186 @@
+func (p *responsemanagementLibraryProxy) getResponsemanagementLibraryById(ctx context.Context, id string) (responsemanagementLibrary *platformclientv2.Library, statusCode int, err error) {
+	return p.getResponsemanagementLibraryByIdAttr(ctx, p, id)
+}
+
+// updateResponsemanagementLibrary updates a Genesys Cloud responsemanagement library
+func (p *responsemanagementLibraryProxy) updateResponsemanagementLibrary(ctx context.Context, id string, responsemanagementLibrary *platformclientv2.Library) (*platformclientv2.Library, error) {
+	return p.updateResponsemanagementLibraryAttr(ctx, p, id, responsemanagementLibrary)
+}
+
+// deleteResponsemanagementLibrary deletes a Genesys Cloud responsemanagement library by Id
+func (p *responsemanagementLibraryProxy) deleteResponsemanagementLibrary(ctx context.Context, id string) (statusCode int, err error) {
+	return p.deleteResponsemanagementLibraryAttr(ctx, p, id)
+}
+
+// createResponsemanagementLibraryFn is an implementation function for creating a Genesys Cloud responsemanagement library
+func createResponsemanagementLibraryFn(ctx context.Context, p *responsemanagementLibraryProxy, responsemanagementLibrary *platformclientv2.Library) (*platformclientv2.Library, error) {
+	library, _, err := p.responseManagementApi.PostResponsemanagementLibraries(*responsemanagementLibrary)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to create responsemanagement library: %s", err)
+	}
+	return library, nil
+}
+
+// getAllResponsemanagementLibraryFn is the implementation for retrieving all responsemanagement library in Genesys Cloud
+func getAllResponsemanagementLibraryFn(ctx context.Context, p *responsemanagementLibraryProxy) (*[]platformclientv2.Library, error) {
+	var allLibrarys []platformclientv2.Library
+	const pageSize = 100
+
+	librarys, _, err := p.responseManagementApi.GetResponsemanagementLibraries(1, pageSize, "", "")
+	if err != nil {
+		return nil, fmt.Errorf("Failed to get library: %v", err)
+	}
+	if librarys.Entities == nil || len(*librarys.Entities) == 0 {
+		return &allLibrarys, nil
+	}
+	for _, library := range *librarys.Entities {
+		allLibrarys = append(allLibrarys, library)
+	}
+
+	for pageNum := 2; pageNum <= *librarys.PageCount; pageNum++ {
+		librarys, _, err := p.responseManagementApi.GetResponsemanagementLibraries(1, pageSize, "", "")
+		if err != nil {
+			return nil, fmt.Errorf("Failed to get library: %v", err)
+		}
+
+		if librarys.Entities == nil || len(*librarys.Entities) == 0 {
+			break
+		}
+
+		for _, library := range *librarys.Entities {
+			allLibrarys = append(allLibrarys, library)
+		}
+	}
+	return &allLibrarys, nil
+}
+
+// getResponsemanagementLibraryIdByNameFn is an implementation of the function to get a Genesys Cloud responsemanagement library by name
+func getResponsemanagementLibraryIdByNameFn(ctx context.Context, p *responsemanagementLibraryProxy, name string) (id string, retryable bool, err error) {
+	librarys, err := getAllResponsemanagementLibraryFn(ctx, p)
```

Comment:
It could be worth adding the prefix param to this function and passing it to `GetResponsemanagementLibraries` , then you could pass `name` to  `getAllResponsemanagementLibraryFn` as the prefix. It could reduce the amount of API calls done if there are loads and loads of libraries out there in the org